samza-api/src/main/java/org/apache/samza/startpoint/Startpoint.java

```diff
@@ -56,7 +55,7 @@ public long getCreationTimestamp() {
    * @param startpointVisitor the visitor of the startpoint.
    * @return the resolved offset.
    */
-  public abstract String apply(SystemStreamPartition systemStreamPartition, StartpointVisitor startpointVisitor);
+  public abstract String apply(SystemStreamPartition systemStreamPartition, StartpointVisitor<SystemStreamPartition, String> startpointVisitor);
```

Comment:
This should also be generified. Otherwise, the `StartpointVisitor` can't be used generically.
e.g.:
`public abstract OUT apply(IN in, StartpointVisitor<IN, OUT> startpointVisitor)`

Reply:
I was quite aware that we'd to generify the startpoint POJO. Without doing that, generifying the startpoint visitor would be pointless. I was in the middle of making the change and you'd reviewed the patch when it was not ready yet.